services/order.py

```diff
@@ -26,8 +30,8 @@ def create_order(
         return order
 
 
-def get_orders(username: str = None) -> QuerySet:
+def get_orders(username: str | None = None) -> QuerySet:
+    filters = {}
     if username:
-        user = User.objects.get(username=username)
-        return Order.objects.filter(user=user)
-    return Order.objects.all()
+        filters["user__username"] = username
+    return Order.objects.filter(**filters)
```

Comment:
You can implement it in 1 line. Each order has a user field that has related_name for Order model. Thus you can find user with that username and use related_name to get all user's orders 👀